### PR TITLE
Remove extra stages on jenkins

### PR DESCRIPTION
**Pull request by pfultz2**

### CODE REVIEW

**Review by causten — APPROVED** [review]

**Comment by pfultz2** [comment]
So hiprtc is broken, so I switched to using hip clang for release and hiprtc for debug.

File: Jenkinsfile [hunk 1]
```diff
@@ -109,21 +109,13 @@ rocmtest clang_debug: rocmnode('vega') { cmake_build ->
...
         def sanitizers = "undefined"
         def debug_flags = "-g -O2 -fsanitize=${sanitizers} -fno-sanitize-recover=${sanitizers}"
-        cmake_build(flags: "-DCMAKE_BUILD_TYPE=debug -DMIGRAPHX_ENABLE_PYTHON=Off -DCMAKE_CXX_FLAGS_DEBUG='${debug_flags}' -DCMAKE_C_FLAGS_DEBUG='${debug_flags}'")
-    }
-}, clang_gpu_debug: rocmnode('vega') { cmake_build ->
-    stage('Hip Clang GPU Debug') {
-        cmake_build(flags: "-DCMAKE_BUILD_TYPE=release", gpu_debug: true)
+        cmake_build(flags: "-DCMAKE_BUILD_TYPE=debug -DMIGRAPHX_ENABLE_PYTHON=Off -DCMAKE_CXX_FLAGS_DEBUG='${debug_flags}' -DCMAKE_C_FLAGS_DEBUG='${debug_flags}' -DMIGRAPHX_USE_HIPRTC=On", gpu_debug: true, hiprtc_workarounds: true)
     }
 }, clang_release: rocmnode('vega') { cmake_build ->
```
Comment (umangyadav) [inline Jenkinsfile:114]: Do we necessary want to keep it on `vega` or we can lable it as `anygpu`.

File: Jenkinsfile [hunk 2]
```diff
@@ -109,21 +109,13 @@ rocmtest clang_debug: rocmnode('vega') { cmake_build ->
     stage('Hip Clang Debug') {
```
Comment (umangyadav) [inline Jenkinsfile:109]: Change name to hipRTC debug

File: Jenkinsfile [hunk 3]
```diff
@@ -109,21 +109,13 @@ rocmtest clang_debug: rocmnode('vega') { cmake_build ->
...
         def debug_flags = "-g -O2 -fsanitize=${sanitizers} -fno-sanitize-recover=${sanitizers}"
-        cmake_build(flags: "-DCMAKE_BUILD_TYPE=debug -DMIGRAPHX_ENABLE_PYTHON=Off -DCMAKE_CXX_FLAGS_DEBUG='${debug_flags}' -DCMAKE_C_FLAGS_DEBUG='${debug_flags}'")
-    }
-}, clang_gpu_debug: rocmnode('vega') { cmake_build ->
-    stage('Hip Clang GPU Debug') {
-        cmake_build(flags: "-DCMAKE_BUILD_TYPE=release", gpu_debug: true)
+        cmake_build(flags: "-DCMAKE_BUILD_TYPE=debug -DMIGRAPHX_ENABLE_PYTHON=Off -DCMAKE_CXX_FLAGS_DEBUG='${debug_flags}' -DCMAKE_C_FLAGS_DEBUG='${debug_flags}' -DMIGRAPHX_USE_HIPRTC=On", gpu_debug: true, hiprtc_workarounds: true)
     }
 }, clang_release: rocmnode('vega') { cmake_build ->
-    stage('Hip Clang Release') {
```
Comment (umangyadav) [inline Jenkinsfile:119]: Keep it HIP clang release ?

File: Jenkinsfile (hunk 1 above)
Comment (pfultz2) [reply Jenkinsfile:114]: Yea

**Review by umangyadav — APPROVED** [review]

File: Jenkinsfile [hunk 4]
```diff
@@ -105,30 +105,22 @@ def rocmnode(name, body) {
     }
 }
 
-rocmtest clang_debug: rocmnode('vega') { cmake_build ->
-    stage('Hip Clang Debug') {
+rocmtest clang_debug: rocmnode('cdna') { cmake_build ->
```
Comment (umangyadav) [inline Jenkinsfile:108]:
```suggestion
rocmtest hiprtc_debug: rocmnode('cdna') { cmake_build ->
```

**Pull request merged**